### PR TITLE
Remove k-NN component from manifest to unblock bundle-build workflow

### PR DESCRIPTION
**Pull request by zelinh**
Signed-off-by: Zelin Hao <zelinhao@amazon.com>

### Description
Our bundle-build job on Jenkins has been failed for a while because of the k-NN plugin and its new docker image. We temporarily remove k-NN component from 1.2.0 manifest to unblock the bundle-build workflow on Jenkins. 
 
### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/822
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaiksaya** [comment]
I am worried about that we would be producing false positive builds. Like build will pass but the cluster wont start because of [knn config in install.sh](https://github.com/opensearch-project/opensearch-build/blob/main/scripts/legacy/tar/linux/opensearch-tar-install.sh#L45)
What do you guys think @peternied @dblock ?

**Comment by peterzhuamazon** [comment]
I think @jmazanec15  is nearly there with his fix, we have confirmed cmake is working properly this morning and he is checking the issues related to `make`. 

Let us see what he says about the progress.

Thanks.

**Review by peternied — APPROVED** [review]
I'd rather unblock the build and then we can address the issues with starting up the cluster separately.

**Review by gaiksaya — APPROVED** [review]

**Comment by zelinh** [comment]
I think @peterzhuamazon is publishing the new staging docker image with the fix on k-NN component so we block this merge for few hours and see if it works.

**Comment by peterzhuamazon** [comment]
Close this as Jack as fixed the issues in #836 #841.

**Pull request closed without merging**
